pkg/util/tls/tls.go

```diff
@@ -29,18 +30,15 @@ func (cfg *ClientConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet)
 	f.StringVar(&cfg.CertPath, prefix+".tls-cert-path", "", "Path to the client certificate file, which will be used for authenticating with the server. Also requires the key path to be configured.")
 	f.StringVar(&cfg.KeyPath, prefix+".tls-key-path", "", "Path to the key file for the client certificate. Also requires the client certificate to be configured.")
 	f.StringVar(&cfg.CAPath, prefix+".tls-ca-path", "", "Path to the CA certificates file to validate server certificate against. If not set, the host's root CA certificates are used.")
+	f.StringVar(&cfg.ServerName, prefix+".tls-server-name", "", "Override the expected name on the server certificate.")
 	f.BoolVar(&cfg.InsecureSkipVerify, prefix+".tls-insecure-skip-verify", false, "Skip validating server certificate.")
 }
 
 // GetTLSConfig initialises tls.Config from config options
 func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
-	// no tls config given at all
-	if cfg.CertPath == "" && cfg.KeyPath == "" && cfg.CAPath == "" {
```

Comment:
Isn't this a breaking change? What's the impact of this change for components that are directly using `ClientConfig`? Before we were retuning `nil` (TLS disabled) while now we do return `tls.Config{}` (TLS enabled).

Reply:
My understanding is that for clients using `tls.Dial` which includes HTTP clients, the behaviour is the exact same if it is set to `nil` or `&tls.Config{}`:

https://github.com/golang/go/blob/master/src/crypto/tls/tls.go#L201